Model/FolderSong.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.IO;
+
+using Claunia.PropertyList;
+using Jammit.Audio;
+
+namespace Jammit.Model
+{
+  /// <summary>
+  /// Represents a song in a raw file sistem folder.
+  /// </summary>
+  class FolderSong : ISong
+  {
+    public SongMeta Metadata { get; }
+
+    public IReadOnlyList<Track> Tracks { get; }
+
+    public IReadOnlyList<Beat> Beats { get; }
+
+    public IReadOnlyList<Section> Sections { get; }
+
+    public FolderSong(SongMeta metadata)
+    {
+      Metadata = metadata;
+      Tracks = InitTracks();
+      Beats = InitBeats();
+      Sections = InitSections();
+    }
+
+    public sbyte[] GetWaveform()
+    {
+      var path = string.Format("{0}/{1}.jcf/music.waveform", Properties.Settings.Default.TrackPath, Metadata.GuidString);
```

Comment:
This and every other instance where you have `TrackPath/Id.jcf` should use the new `SongPath` field I suggested in other comments.